File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/loader/TempLocalItem.java
package ru.kmorozov.gbd.core.loader;

import ru.kmorozov.gbd.core.logic.model.book.base.IPage;

import java.io.File;
import java.io.IOException;

public class TempLocalItem extends LocalFSStoredItem {

    public TempLocalItem(LocalFSStorage storage, IPage page, String imgFormat) throws IOException {
        super(File.createTempFile(page.getOrder() + '_' + page.getPid(), imgFormat), storage, page, imgFormat);
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/loader/TempLocalItem.kt
package ru.kmorozov.gbd.core.loader

import ru.kmorozov.gbd.core.logic.model.book.base.IPage

import java.io.File
import java.io.IOException

class TempLocalItem @Throws(IOException::class)
constructor(storage: LocalFSStorage, page: IPage, imgFormat: String) : MayBePageItem(File.createTempFile(page.order.toString() + '_'.toString() + page.pid, imgFormat), page)
